pulpcore/pulpcore/app/models/publication.py

```diff
@@ -102,3 +104,12 @@ class Distribution(Model):
     class Meta:
         unique_together = ('publisher', 'name')
         default_related_name = 'distributions'
+
+    def get_base_url(self):
+        """
+        Helper method for constructing the base_url field
+
+        Returns:
+            URL that can be used for consuming content distributed by this distribution
+        """
+        return ''.join([settings.CONTENT['hostname'], reverse('content-app'), self.base_path])
```

Comment:
@jortel I don't have access to the request object here. So I am not sure how I can get the host without a setting. Do you have any suggestions?

Reply:
This should be handled by a serializer, similar to what we did with the CreatedResourceSerializer.to_representation. Models shouldn't know about their urls.